File: timetrials/src/main/java/fzero/timetrials/controller/SubmissionController.java
package fzero.timetrials.controller;

import fzero.timetrials.SubmissionApplicationService;
import fzero.timetrials.dto.SubmissionDto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/submission")
public class SubmissionController {
    @Autowired
    private SubmissionApplicationService submissionApplicationService;

    @GetMapping("")
    public String getSubmitPage(Model model) {
        model.addAttribute("submissionDto", new SubmissionDto());
        return "submission";
    }

    @PostMapping("/new")
    public String submitTime(@ModelAttribute("submissionDto") SubmissionDto submissionDto) {
        submissionApplicationService.saveRun(submissionDto);
        return "redirect:/submission";
    }
}

File: FZero99-TT/src/main/kotlin/fZero/fZero99TT/controller/SubmissionController.kt
package fZero.fZero99TT.controller

import fZero.fZero99TT.applicationService.SubmissionApplicationService
import fZero.fZero99TT.dto.SubmissionDto
import fZero.fZero99TT.game.FZeroGlider
import fZero.fZero99TT.game.FZeroTrack
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/submission")
class SubmissionController(
        private val submissionApplicationService: SubmissionApplicationService,
) {
    @GetMapping("")
    fun getSubmitPage(model: Model): String {
        model.addAttribute("submissionDto", SubmissionDto())
        model.addAttribute("trackChoices", FZeroTrack.values())
        model.addAttribute("gliderChoices", FZeroGlider.values())
        return "submission"
    }

    @PostMapping("/new")
    fun submitTime(@ModelAttribute("submissionDto") submissionDto: SubmissionDto): String {
        submissionApplicationService.saveRun(submissionDto)
        return "redirect:/submission"
    }
}
